drivers/spi/spi_ll_stm32.c

```diff
@@ -250,6 +250,71 @@ static int spi_stm32_get_err(SPI_TypeDef *spi)
 	return 0;
 }
 
+static void spi_stm32_shift_half_m(SPI_TypeDef *spi,
+				   struct spi_stm32_data *data)
+{
+	if (spi_context_tx_buf_on(&data->ctx)) {
+		uint16_t tx_frame;
+
+		if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
+			tx_frame = UNALIGNED_GET((uint8_t *)(data->ctx.tx_buf));
+			LL_SPI_TransmitData8(spi, tx_frame);
+			spi_context_update_tx(&data->ctx, 1, 1);
+		} else {
+			tx_frame = UNALIGNED_GET((uint16_t *)(data->ctx.tx_buf));
+			LL_SPI_TransmitData16(spi, tx_frame);
+			spi_context_update_tx(&data->ctx, 2, 1);
+		}
+
+		if (!spi_context_tx_buf_on(&data->ctx) &&
+		    spi_context_rx_buf_on(&data->ctx)) {
+			while (!ll_func_tx_is_empty(spi)) {
+				/* NOP */
+			}
+
+			while (ll_func_spi_is_busy(spi)) {
+				/* NOP - Waiting until TX is done */
+			}
+
+			while (ll_func_rx_is_not_empty(spi)) {
+				/* Cleaning up RX */
+				LL_SPI_ReceiveData8(spi);
+			}
+
+			/* Switching to RX */
+			LL_SPI_SetTransferDirection(spi, LL_SPI_HALF_DUPLEX_RX);
```

Comment:
I think maybe the only thing required here is to let the user do a call to `spi_write` and then another call to `spi_read` (instead of transceive). With that, I think we only need to support to configure the SPI peripheral as full-duplex or half-duplex as a function of the SPI cpnfiguration received, and let the user do 2 separated calls (with the appropriate config.) to write and read.